### PR TITLE
Prevent NPE from being thrown by the `export` sub-command if `testFramework` isn't defined

### LINKED ISSUE #1765: NPE while exporting to sbt or mill
**Version(s)**
0.1.19

**Describe the bug**
Exporting to sbt fails with NPE trying to infer a test framework name

**To Reproduce**
Running the export command `scala-cli export --sbt server.scala -o sbt-proj` on this file 

<details>
<summary>server.scala</summary>

```scala
//> using scala "3.2.1"
//> using resourceDir "."
//> using packaging.packageType "assembly"
//> using lib "org.http4s::http4s-ember-server::0.23.17"
//> using lib "org.http4s::http4s-dsl::0.23.17"
//> using lib "com.monovore::decline-effect::2.4.1"
//> using lib "ch.qos.logback:logback-classic:1.4.5"

import cats.effect.{ExitCode, IO}
import cats.effect.kernel.Async
import cats.syntax.all.*
import com.comcast.ip4s.{ipv4, port}
import com.monovore.decline.Opts
import com.monovore.decline.effect.CommandIOApp
import org.http4s.{HttpRoutes, MediaType, Response, Status, Uri}
import org.http4s.dsl.io.*
import org.http4s.ember.server.EmberServerBuilder
import org.http4s.headers.`Content-Type`
import org.http4s.server.middleware.CORS
import org.typelevel.log4cats.Logger
import org.typelevel.log4cats.slf4j.*

object Server extends CommandIOApp("helloServer", "Titles you in HTML") {

  val titleOpt: Opts[String] =
    Opts.env[String]("TITLE", "Page title").withDefault("Hello")

  val baseUrlOpt: Opts[Uri] = Opts
    .env[String]("BASE_URL", "The base url")
    .mapValidated(
      Uri
        .fromString(_)
        .leftMap(_.message)
        .ensure("base url must be absolute")(_.path.addEndsWithSlash.absolute)
        .map(uri => uri.withPath(uri.path.dropEndsWithSlash))
        .toValidatedNel
    )

  def page(uri: Uri, title: String): String =
    s"""|<html>
        |<head><title>$title</title></head>
        |<body>Hello from ${uri.toString}</body>
        |</html>""".stripMargin

  def routes[F[_]: Async: Logger](baseUrl: Uri, title: String): HttpRoutes[F] =
    HttpRoutes.of[F] {
      case GET -> Root / "health" => Response[F](Status.Ok).pure[F]
      case GET -> path =>
        Logger[F].info(s"Serving $path") >>
          Response[F](Status.Ok)
            .withEntity(page(baseUrl.withPath(baseUrl.path.merge(path)), title))
            .withContentType(`Content-Type`(MediaType.text.html))
            .pure[F]
    }

  def main: Opts[IO[ExitCode]] = (baseUrlOpt, titleOpt).mapN((baseUrl, title) =>
    for {
      given Logger[IO] <- Slf4jFactory.create[IO]
      exitCode <- EmberServerBuilder
        .default[IO]
        .withHttp2
        .withHost(ipv4"0.0.0.0")
        .withPort(port"8080")
        .withHttpApp(
          CORS.policy.withAllowOriginAll(routes[IO](baseUrl, title)).orNotFound
        )
        .build
        .useForever
        .as(ExitCode.Success)
    } yield exitCode
  )
}
```
</details>

causes scala-cli to fail with the following exception:

<details>
<summary>stacktrace</summary>

```
java.lang.NullPointerException
  java.util.concurrent.ConcurrentHashMap.get(ConcurrentHashMap.java:936)
  scala.build.testrunner.AsmTestRunner$ParentInspector.parents(AsmTestRunner.scala:20)
  scala.build.testrunner.AsmTestRunner$ParentInspector.$anonfun$allParents$1(AsmTestRunner.scala:44)
  scala.collection.immutable.Stream$Cons.tail(Stream.scala:403)
  scala.collection.immutable.Stream$Cons.tail(Stream.scala:394)
  scala.collection.LinearSeqOps.contains(LinearSeq.scala:165)
  scala.collection.LinearSeqOps.contains$(LinearSeq.scala:161)
  scala.collection.immutable.Stream.contains(Stream.scala:29)
  scala.build.testrunner.AsmTestRunner$.$anonfun$findFramework$6(AsmTestRunner.scala:238)
  scala.collection.Iterator$$anon$10.nextCur(Iterator.scala:594)
  scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:608)
  scala.collection.Iterator$SliceIterator.hasNext(Iterator.scala:1240)
  scala.collection.immutable.List.prependedAll(List.scala:152)
  scala.collection.IterableOnceOps.toList(IterableOnce.scala:1288)
  scala.collection.IterableOnceOps.toList$(IterableOnce.scala:1288)
  scala.collection.AbstractIterator.toList(Iterator.scala:1300)
  scala.build.testrunner.AsmTestRunner$.findFramework(AsmTestRunner.scala:244)
  scala.build.internal.Runner$.$anonfun$7(Runner.scala:381)
  scala.Option.orElse(Option.scala:477)
  scala.build.internal.Runner$.frameworkName(Runner.scala:383)
  scala.cli.exportCmd.Sbt.$anonfun$18(Sbt.scala:259)
  scala.Option.orElse(Option.scala:477)
  scala.cli.exportCmd.Sbt.testFrameworkSettings(Sbt.scala:260)
  scala.cli.exportCmd.Sbt.export(Sbt.scala:352)
  scala.cli.exportCmd.Sbt.export(Sbt.scala:324)
  scala.cli.commands.export0.Export$.runCommand(Export.scala:148)
  scala.cli.commands.export0.Export$.runCommand(Export.scala:80)
  scala.cli.commands.ScalaCommand.run(ScalaCommand.scala:273)
  scala.cli.commands.ScalaCommand.run(ScalaCommand.scala:266)
  caseapp.core.app.CaseApp.main(CaseApp.scala:150)
  caseapp.core.app.CommandsEntryPoint.main(CommandsEntryPoint.scala:120)
  scala.cli.ScalaCliCommands.main(ScalaCliCommands.scala:124)
  scala.cli.ScalaCli$.main0(ScalaCli.scala:189)
  scala.cli.ScalaCli$.main(ScalaCli.scala:93)
  scala.cli.ScalaCli.main(ScalaCli.scala)
```
</details>

My wild guess is that scala-cli is failing while parsing the testFramework name [here](https://github.com/VirtusLab/scala-cli/blob/518d63de7592b665c3134b5f2b0905e7a8a4abdc/modules/cli/src/main/scala/scala/cli/exportCmd/Sbt.scala#L258-L260C6).
Adding the directive `//> using testFramework ""` fixes the bug, and the project gets exported correctly. 

Also, I can't export to mill either, so the issue seems to occur while "parsing" the .scala file

Lastly, somehow, scala-cli fails after 5/6 seconds, while exporting the same file with the testFramework directive gets exported in less a second.

**Expected behaviour**
The project should get created :D

### PR DESCRIPTION
**Pull request by Gedochao**
Fixes #1765

### CODE REVIEW

File: modules/integration/src/test/scala/scala/cli/integration/ExportTestProjects.scala [hunk 1]
```diff
@@ -270,4 +270,12 @@ object ExportTestProjects {
          |""".stripMargin
     TestInputs(os.rel / "Test.scala" -> testFile)
   }
+
+  def logbackBugCase(scalaVersion: String): TestInputs =
+    TestInputs(os.rel / "script.sc" ->
+      s"""//> using scala "$scalaVersion"
+         |//> using lib "ch.qos.logback:logback-classic:1.4.5"
+         |///> using testFramework ""
```
Comment (lwronski) [inline modules/integration/src/test/scala/scala/cli/integration/ExportTestProjects.scala]: Remove the commented `using` directive `testFramework`

Comment (Gedochao) [reply modules/integration/src/test/scala/scala/cli/integration/ExportTestProjects.scala]: fixed

File: modules/test-runner/src/main/scala/scala/build/testrunner/AsmTestRunner.scala [hunk 2]
```diff
@@ -16,23 +16,24 @@ object AsmTestRunner {
...
-              val checker = new TestClassChecker
-              reader.accept(checker, 0)
-              checker.implements
-          }
-
-          cache.put(className, parents)
-          parents
-      }
+    private def parents(className: String): Seq[String] =
+      Option(className).map { cn =>
```
Comment (alexarchambault) [inline modules/test-runner/src/main/scala/scala/build/testrunner/AsmTestRunner.scala]: When can `className` be null here? Is it when a null value is passed to `allParents`, or when `helper` in `allParents` iterates over parents? In the former case, the null check could be moved to `allParents`. In the latter case, it seems the stream returned by `allParents` can contain some nulls, which feels odd.

Comment (Gedochao) [reply modules/test-runner/src/main/scala/scala/build/testrunner/AsmTestRunner.scala]: It does seem to be the latter case, as I didn't manage to catch it in `allParents`...

Comment (Gedochao) [reply modules/test-runner/src/main/scala/scala/build/testrunner/AsmTestRunner.scala]: it seems the null was coming from `TestClassChecker.implements`, the returned list could contain nulls.
should be fixed

**Review by lwronski — APPROVED** [review]
LGTM

**Pull request merged**